File: solutions/cpp/series/series.cpp
#include "series.h"

#include <stdexcept>

namespace series {

    auto slice(const std::string &s, int l) -> std::vector<std::string>
    {
        if (l < 1 || l > static_cast<int>(s.length()))
            throw std::domain_error("");

        std::vector<std::string> slices;

        for (std::size_t i = 0; i <= s.length() - l; i++)
        {
            slices.push_back(s.substr(i, l));
        }

        return slices;
    }

}  // namespace series

File: solutions/python/series/series.py
def slices(series, length):
    if series == "":
        raise ValueError("series cannot be empty")
    elif length > len(series):
        raise ValueError("slice length cannot be greater than series length")
    elif length == 0:
        raise ValueError("slice length cannot be zero")
    elif length < 0:
        raise ValueError("slice length cannot be negative")
    else:
        ss = []
        for i in range(0, len(series) - length + 1):
            ss += [series[i : i + length]]
        return ss

